### PR TITLE
Feature add generic list

### LINKED ISSUE #165: Add a `GenericList<T>` interface for multi-value attributes.

### PR DESCRIPTION
**Pull request by izzahaj**
Closes #165 
This PR adds a `GenericList<T>` interface to encapsulate multi-valued attributes of the `Person` class.

### CODE REVIEW

File: src/main/java/seedu/address/model/GenericList.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    /**
+     * Edits an element in the {@code GenericList<T>}.
+     *
+     * @param index The index of the element to be edited.
+     * @return The updated {@code GenericList<T>}.
+     */
+    GenericList<T> edit(int index, T t);
+
+    T get(int index);
```
Comment (Ramanathan0908) [inline src/main/java/seedu/address/model/GenericList.java:34]: Should the index be of type Index instead? then the implementation can decide to use zeroBased or oneBased. Not sure if this is the right way but they are using index class to convey this sort of info right (so the places using these methods does not need to care about computing the index)

**Review by Ramanathan0908 — APPROVED** [review]
A minor comment which I am not sure is needed, would be good if others gave their input. But other than that LGTM

File: src/main/java/seedu/address/model/GenericList.java (hunk 1 above)
Comment (izzahaj) [reply src/main/java/seedu/address/model/GenericList.java:34]: I think the conversion of `Index` to the approriate `int` is already done in the corresponding `Command` classes.

Comment (Ramanathan0908) [reply src/main/java/seedu/address/model/GenericList.java:34]: Okay sure then LGTM

**Pull request merged**